InventoryManagementSystem/Domain/InventoryManagement/Inventory.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using InventoryManagementSystem.Domain.Reader;
+
+namespace InventoryManagementSystem.Domain
+{
+    public class Inventory
+    {
+        public List<Product> products = new List<Product>();
+        private readonly IReader reader;
+
+        public Inventory(IReader reader)
+        {
+           this.reader = reader;
+        }
+        public void AddProduct()
+        {
+            Product product = reader.ReadProduct();
+            products.Add(product);
+        }
+        public void ViewAllProducts()
+        {
+            Console.WriteLine("The Products:");
+            foreach (var product in products)
```

Comment:
this logic can be transferred to the IReader also, could be by creating a new method and passing the products list to that method.